src/plugins/data/public/ui/query_string_input/query_string_input.tsx

```diff
@@ -119,24 +120,34 @@ export default class QueryStringInputUI extends Component<Props, State> {
     return toUser(this.props.query.query);
   };
 
-  private fetchIndexPatterns = async () => {
+  private fetchIndexPatterns = debounce(async () => {
     const stringPatterns = this.props.indexPatterns.filter(
       (indexPattern) => typeof indexPattern === 'string'
     ) as string[];
     const objectPatterns = this.props.indexPatterns.filter(
       (indexPattern) => typeof indexPattern !== 'string'
     ) as IIndexPattern[];
 
+    this.fetchIndexPatternsAbortController = new AbortController();
+
     const objectPatternsFromStrings = (await fetchIndexPatterns(
       this.services.savedObjects!.client,
       stringPatterns,
       this.services.uiSettings!
     )) as IIndexPattern[];
 
-    this.setState({
-      indexPatterns: [...objectPatterns, ...objectPatternsFromStrings],
-    });
-  };
+    if (!this.fetchIndexPatternsAbortController.signal.aborted) {
```

Comment:
I didn't test but, if I understood this correctly, this doesn't seem to be working as intended 🤔 
Currently every time we are checking the LATEST abort controller, but we should instead check to the 
abort controller which was created when we fired a corresponding fetch:

I think it should look something like this instead:
```
if (this.lastAbortController) this.lastAbortController.abort(); // make sure any older requests are canceled.
this.lastAbortController = new AbortController();
const currentAbortController = this.lastAbortController;
await fetch();
if (!currentAbortController.signal.aborted) {
 // 
 this.setState()
 this.updateSuggestions
}

```

Reply:
Thanks @Dosant , updated